examples/cifar10/test.cpp

```diff
@@ -22,16 +18,14 @@ void convert_image(const std::string &imagefilename,
                    int w,
                    int h,
                    vec_t &data) {
-  cv::Mat img = cv::imread(imagefilename);
-  if (img.data == nullptr) return;  // cannot open, or it's not an image
-  cv::Mat resized;
-  cv::resize(img, resized, cv::Size(w, h), .0, .0);
-  data.resize(w * h * resized.channels(), minv);
-  for (int c = 0; c < resized.channels(); ++c) {
-    for (int y = 0; y < resized.rows; ++y) {
-      for (int x = 0; x < resized.cols; ++x) {
+  image<> img(imagefilename, image_type::rgb);
+  image<> resized = resize_image(img, w, h);
+  data.resize(w * h * 3);
```

Comment:
Yeah, that 3 is ugly but I was't sure what's the best approach to handle it. 
1. Define a macro? #define N_CHANNELS 3 // number of channels
2. Define int depth = 3; and plug 3 in there? // number of channels
I did look at the line you referenced earlier but it was a private variable. We could define a get_channel() method perhaps?